src/main/java/ch/admin/bag/covidcertificate/service/COSEService.java

```diff
@@ -32,15 +32,15 @@ public byte[] getCOSESign1(byte[] dgcCBOR, SigningInformation signingInformation
 
     private byte[] getProtectedHeader(SigningInformation signingInformation) {
         try {
-            //TODO: keyIdentifier should be deleted. It is deprecated and used only for backwards compatibility.
-            var keyIdentifier = signingInformation.getKeyIdentifier();
-            if(signingInformation.getCertificateAlias() != null && !signingInformation.getCertificateAlias().isBlank()){
-                keyIdentifier = signingClient.getKeyIdentifier(signingInformation.getCertificateAlias());
+            if (signingInformation.getCertificateAlias() != null &&
+                    !signingInformation.getCertificateAlias().isBlank()) {
+                var keyIdentifier = signingClient.getKeyIdentifier(signingInformation.getCertificateAlias());
```

Comment:
```
    private byte[] getProtectedHeader(SigningInformation signingInformation) {
        try {
            if (StringUtils.isNotBlank(signingInformation.getCertificateAlias())) {
                var keyIdentifier = signingClient.getKeyIdentifier(signingInformation.getCertificateAlias());
                return cborService.getProtectedHeader(keyIdentifier);
            }
        } catch (Exception e) {
            throw new CreateCertificateException(CREATE_COSE_PROTECTED_HEADER_FAILED);
        }
        throw new CreateCertificateException(CREATE_COSE_PROTECTED_HEADER_FAILED);
    }
```

Reply:
+1 for the suggested use of `StringUtils.isNotBlank`

Reply:
Done mit org.apache.commons.lang3.StringUtils